### PR TITLE
Allow --module node16

### PR DESCRIPTION
**Pull request by andrewbranch**
Related: https://github.com/DefinitelyTyped/dt-mergebot/pull/458

### CODE REVIEW

File: packages/dtslint/src/checks.ts [hunk 1]
```diff
@@ -56,16 +56,24 @@ export async function checkPackageJson(dirPath: string, typesVersions: readonly
   }
 }
 
+/**
+ * numbers in `CompilerOptions` might be enum values mapped from strings
```
Comment (sandersn) [inline packages/dtslint/src/checks.ts:60]: which specific properties have this? I assume it's "module".

File: packages/dtslint/src/checks.ts [hunk 2]
```diff
@@ -56,16 +56,24 @@ export async function checkPackageJson(dirPath: string, typesVersions: readonly
...
+ */
+export type CompilerOptionsRaw = {
+  [K in keyof CompilerOptions]?: CompilerOptions[K] extends number | undefined
+    ? string | number | undefined
```
Comment (sandersn) [inline packages/dtslint/src/checks.ts:64]: do properties normally get literal union types instead of `string | number | undefined` ?

**Review by sandersn — APPROVED** [review]
Looks good:
1. A couple of questions about the new CompilerOptionsRaw type, although they're incidental.
2. I want to understand why all packages should eventually switch to node16, although shipping the ability for some to be node16 isn't worth waiting on that.

File: packages/dtslint/src/checks.ts (hunk 1 above)
Comment (andrewbranch) [reply packages/dtslint/src/checks.ts:60]: `module` is the only one that matters here, but there are other enums that aren’t relevant to DT. Most numbers in `CompilerOptions` are actually enums represented by strings in the raw form. The only real number I can think of is `maxNodeModulesJsDepth`.

File: packages/dtslint/src/checks.ts (hunk 2 above)
Comment (andrewbranch) [reply packages/dtslint/src/checks.ts:64]: Most of the ones that are getting transformed here are enum types, which are basically literal union types.

**Pull request merged**